tests/unittests/Foundation/CoreFoundationTests.m

```diff
@@ -56,3 +57,19 @@ static uint64_t walltime(const struct mach_timebase_info& timebase, uint64_t val
     EXPECT_GT(measurement3, measurement2);
     EXPECT_GT(measurement2, measurement1);
 }
+
+TEST(CFType, UnknownTypeInObjCContainer) {
+    // This test requires a non-bridged public CF type. CFTreeRef will suffice.
+    CFTreeContext treeContext{};
+    CFTypeRef tree = static_cast<CFTypeRef>(CFTreeCreate(kCFAllocatorDefault, &treeContext));
+
+    NSArray* array = nil;
+
+    // The ',' in this method call upsets EXPECT_NO_THROW(...), so we need to hide it behind a lambda.
+    auto create = [&tree](){
+        return [[NSArray alloc] initWithObjects:static_cast<id>(tree), nil];
+    };
+    EXPECT_NO_THROW(array = create());
```

Comment:
`EXPECT_*_THROW` are ARM-safe; it's `@catch(NSException*)` that's not ARM-safe :)

---
In reply to: [81263134](https://github.com/Microsoft/WinObjC/pull/1059#discussion_r81263134) [](ancestors = 81263134)